### PR TITLE
Add Go sample project for many in-flight two-phase transfers

### PR DESCRIPTION
**Pull request by eatonphil**
Part of https://github.com/tigerbeetle/tigerbeetle/issues/943.

### CODE REVIEW

File: src/clients/go/samples/two-phase-many/main.go [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+		{
+			ID: uint128("2"),
+			DebitsPosted: uint64(0),
+			CreditsPosted: uint64(100),
+			DebitsPending: uint64(0),
+			CreditsPending: uint64(1400),
+		},
+	}, "after completing 1 pending transfer")
+
+	// Create a 7th transfer voiding the 2nd transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Create a 7th transfer voiding the 2nd transfer.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+		{
+			ID: uint128("2"),
+			DebitsPosted: uint64(0),
+			CreditsPosted: uint64(0),
+			DebitsPending: uint64(0),
+			CreditsPending: uint64(1500),
+		},
+	}, "after starting 5 pending transfers")
+
+	// Create a 6th transfer posting the 1st transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Create a 6th transfer posting the 1st transfer.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+		port = "3000"
+	}
+
+	client, err := tb.NewClient(0, []string{port}, 32)
+	if err != nil {
+		log.Fatalf("Error creating client: %s", err)
+	}
+	defer client.Close()
+
+	// Create two accounts
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Create two accounts.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+	})
+	if err != nil {
+		log.Fatalf("Error creating accounts: %s", err)
+	}
+
+	for _, err := range res {
+		log.Fatalf("Error creating account %d: %s", err.Index, err.Result)
+	}
+
+	// Start five pending transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Start five pending transfer.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+	transferRes, err := client.CreateTransfers(transfers)
+	if err != nil {
+		log.Fatalf("Error creating transfer: %s", err)
+	}
+
+	for _, err := range transferRes {
+		log.Fatalf("Error creating transfer: %s", err.Result)
+	}
+
+	// Validate accounts pending and posted debits/credits before finishing the two-phase transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Validate accounts pending and posted debits/credits before finishing the two-phase transfer.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+		{
+			ID: uint128("2"),
+			DebitsPosted: uint64(0),
+			CreditsPosted: uint64(100),
+			DebitsPending: uint64(0),
+			CreditsPending: uint64(1200),
+		},
+	}, "after completing 2 pending transfers")
+
+	// Create a 8th transfer posting the 3rd transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Create a 8th transfer posting the 3rd transfer.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+		{
+			ID: uint128("2"),
+			DebitsPosted: uint64(0),
+			CreditsPosted: uint64(400),
+			DebitsPending: uint64(0),
+			CreditsPending: uint64(900),
+		},
+	}, "after completing 3 pending transfers")
+
+	// Create a 9th transfer voiding the 4th transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Create a 9th transfer voiding the 4th transfer.
```

File: src/clients/go/samples/two-phase-many/main.go [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+		{
+			ID: uint128("2"),
+			DebitsPosted: uint64(0),
+			CreditsPosted: uint64(400),
+			DebitsPending: uint64(0),
+			CreditsPending: uint64(500),
+		},
+	}, "after completing 4 pending transfers")
+
+	// Create a 10th transfer voiding the 5th transfer
```
Comment (batiati) [inline src/clients/go/samples/two-phase-many/main.go]:
```suggestion
	// Create a 10th transfer voiding the 5th transfer.
```

**Review by batiati — CHANGES_REQUESTED** [review]
Thanks @eatonphil!

Just taking the opportunity to write the comments following the TigerStyle standard.
Most of the non-zig code doesn't follow this convention ... for no reason 🤷‍♂️.

**Comment by eatonphil** [comment]
Thanks @batiati yeah I just noticed this myself as well as I am porting this project to Node.

**Review by batiati — APPROVED** [review]
:+1:

**Pull request merged**